api/preProcessor-template.ts

```diff
@@ -0,0 +1,19 @@
+import thing from 'module';
+
+export let sourceExtensionTypes: string[] = [".ts", ".tsx"];
+
+export let outputExtensionTypes: string[] = [".mjs", ".cjs"];
+
+
+// Example of defining "options" for a TypeScript loader
+export interface TypeScriptOptions {
+  tsconfigFile?: string
+}
+
+export function process(source: string | Buffer, options: TypeScriptOptions = {}): string| Buffer {
+
+  // Example of returning a transpiled source using options provided by project config
```

Comment:
This currently omits one interesting bit: Above you declare that it may return either `.mjs` or `.cjs` but the return value doesn't include which one the returned source text represents.

Reply:
Good point. Overall, I've been thinking of making output extensions more concrete as I delve into the "build" use case for PRESM tool.  Addressing this now, however, maybe outputting an `object` with a `source` and `extension` as you mentioned below makes sense.